src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -3250,6 +3250,92 @@ class AlterStageSegment(BaseSegment):
     )
 
 
+@snowflake_dialect.segment()
+class AppendOnlyExpressionSegment(BaseSegment):
+    """An APPEND_ONLY = TRUE | FALSE expression."""
+
+    type = "append_only_expression"
+    
+    match_grammar = Sequence(
+        "APPEND_ONLY",
+        Ref("EqualsSegment"),
+        OneOf(Ref("TrueSegment"), Ref("FalseSegment"))
+    )
+
+
+@snowflake_dialect.segment()
+class ShowInitialRowsExpressionSegment(BaseSegment):
+    """A SHOW_INITIAL_ROWS = TRUE | FALSE expression."""
+
+    type = "show_initial_rows_expression"
+
+    match_grammar = Sequence(
+        "SHOW_INITIAL_ROWS",
+        Ref("EqualsSegment"),
+        OneOf(Ref("TrueSegment"), Ref("FalseSegment"))
+    )
+
+
+@snowflake_dialect.segment()
+class InsertOnlyExpressionSegment(BaseSegment):
+    """A INSERT_ONLY = TRUE expression."""
+
+    type = "insert_only_expression"
+
+    match_grammar = Sequence(
+        "INSERT_ONLY",
+        Ref("EqualsSegment"),
+        Ref("TrueSegment")
+    )
```

Comment:
There's a question as to whether these need to be separate statements or should just all be directly part of `CreateStreamStatementSegment`.

Having separate statements can be useful if:
1. Same code is used in multiple places
2. Want to be able to give it a `type` so can reference it in API or in rules.
3. Breaks up long code (though counter argument is you can't understand everything in one place).

I don't think 1 and 2 apply here and I could fall either way on 3 in this case. Is there a reason you think they are better as separate segments?

Reply:
I'm inclined to agree. I observed elsewhere that most segments were broken out to make it `DRY`, so I thought I'd do the same. However, I've only seen these statements written in this syntax as part of the STREAM object. 
I'll search the snowflake docs to double-check.

One after-thought is that the ALTER statement (which I haven't written into this PR but might choose to if I have the cycles) might reuse these statements.

Reply:
Well happy to keep them if you think they might be reused. As I say, I'm on the fence about them, rather than thinking they are wrong.

Alternatively they are relatively simple so, even if ALTER statement needs them, could just repeat them. Do want to DRY code for more complex statements, but for simple things a little repetition isn't the worst and has benefits.

Anyway, let's get rest of your tests fixed first and then can come back to this and decide one way or the other.

Reply:
I've opted to directly integrate them. At I was trying to resolve what turned out to be a misspelled reference (`InsertOnlyExpressionStatement` vs. `InsertOnlyExpressionSegment`), but then kept them out because of our dialog.